authorizers/iconik/headers-authorizer.ts

```diff
@@ -0,0 +1,33 @@
+import { getEnv } from '@helper/environment';
+import { errorHandler } from '@helper/error-handler';
+import { log } from '@helper/logger';
+import { IconikParams } from '@workflowwin/iconik-api';
+import { APIGatewayAuthorizerEvent, Handler } from 'aws-lambda';
+import { getCallerAndOwner } from './helper';
+import { generatePolicy } from '../policy-generator';
+import { AuthorizerResponse, IconikContext, IconikEnhancedAuthContext } from './interfaces/context';
+
+export const iconikAuthorizer: Handler<APIGatewayAuthorizerEvent, AuthorizerResponse> = async (event, context) => {
+  log('[Iconik Authorizer]', event);
+  try {
+    const callerId: string = event['headers']['User-Id'];
+    const authToken: string = event['headers']['auth-token'];
+
+    const iconikUrl = getEnv('ICONIK_URL');
+    const appId = getEnv('ICONIK_APP_ID');
+    const systemDomainId = getEnv('ICONIK_DOMAIN_ID');
+
+    const iconikParams: IconikParams = { authToken, iconikUrl, appId, systemDomainId };
+
+    const { appOwner }: IconikContext = await getCallerAndOwner(iconikParams, callerId);
+    const context: IconikEnhancedAuthContext = {
+      ...iconikParams,
+      ownerId: appOwner?.id,
+      ownerEmail: appOwner?.email,
+    };
+
+    return generatePolicy<IconikEnhancedAuthContext | any>(`user|${appOwner?.id}`, 'Allow', event.methodArn, context);

```

Comment:
I see the issue here. The polyce will be cached for `user|${appOwner?.id}` and when user call another lambda(with another ARN) he get deny response for that method. You have the following options to avoid that:
- Off cache. It afffect perfomance. 
- Provide '*' as arn. The most simple way, but has own disadvantages.
- Provide an ARN pattern(as far as i remembre it`s possible) or list of ARNs.

Reply:
For now, I will choose the second way.